app/src/main/java/com/winlator/cmod/XrActivity.java

```diff
@@ -364,6 +365,11 @@ public static void updateControllers() {
                     xrAPI.sendUDP(xrAPI.encodeAxes(lastAxes, 0), XrAPI.DEFAULT_PORT);
                     //xrAPI.sendFile(xrAPI.encodeAxes(lastAxes, 0), 0);
                 }
+
+                isDebugUDP = xrAPI.hasDebugIP();
+                if (isDebugUDP) {
+                    xrAPI.sendDebugUDP(xrAPI.encodeAxes(lastAxes, 0), XrAPI.DEFAULT_PORT);
```

Comment:
I think we should make sensDebugUDP and hasDebugIP private and move the code inside send UDP (that way you don't need to encodeAxes twice and the code would be cleaner).